dragonfly/implementations/roach2interface.py

```diff
@@ -0,0 +1,316 @@
+            self.cfg_list.append(cfg_c)
+            self.channel_list.append('c')
+
+        logger.info('Number of channels: {}'.format(len(self.channel_list)))
+
+
+        ArtooDaq.__init__(self, self.roach2_hostname, boffile=boffile, do_ogp_cal=do_ogp_cal, do_adcif_cal=do_adcif_cal, ifcfg=self.cfg_list)
+        self.configured=True
+
+        if boffile!=None:
+            self.do_adc_ogp_calibration()
+
+        for s in self.channel_list:
+            self.set_central_frequency(self.default_frequency, s)
+            self.gain = (self.gain_dict[s], s)
+            self.fft_shift_vector = (self.fft_shift, 'ab')
+            self.fft_shift_vector = (self.fft_shift, 'cd')
+        return self.configured
+
+
+    def get_calibration_status(self):
+        return self.calibrated
+
+
+    def get_configuration_status(self):
+        return self.configured
+
+
+    def do_adc_ogp_calibration(self, **kwargs):
+        logger.info('Calibrating ROACH2, this will take a while.')
+        logger.info('Doing adc ogp calibration')
+        adc_dictionary = ArtooDaq.calibrate_adc_ogp(self, **kwargs)
+        logger.info(adc_dictionary)
+        self.calibrated=True
+        return self.calibrated
+
+
+    @property
+    def is_running(self):
+        logger.info('Pinging ROACH2')
+        response = os.system("ping -c 1 " + self.roach2_hostname)
+        #and then check the response...
+        if response == 0:
+            logger.info('ROACH2 is switched on')
+
+            if not self.get_configuration_status:
+                logger.info('ROACH2 is not configured')
+            if not self.get_calibration_status:
+                logger.info('Roach2 is not calibrated')
+        else:
+            self.configured=False
+            self.calibrated=False
+
+        return self.calibrated
+
+
+    def block_channel(self, channel):
+        self.block_dict[channel]=True
+
+
+    def unblock_channel(self, channel):
+        self.block_dict[channel]=False
+
+
+    @property
+    def blocked_channels(self):
+        bc = [i for i in self.block_dict.keys() if self.block_dict[i]==True]
+        return bc
+
+
+    def get_central_frequency(self, channel):
+        return self.freq_dict[channel]
+
+
+    def set_central_frequency(self, cf, channel):
+        if self.block_dict[channel]==False:
+            logger.info('setting central frequency of channel {} to {}'.format(channel, cf))
+            cf = ArtooDaq.tune_ddc_1st_to_freq(self, cf, tag=channel)
+            self.freq_dict[channel]=cf
+            return cf
+        else:
+            raise core.exceptions.DriplineGenericDAQError('Channel {} is blocked'.format(channel))
+
+
+    @property
+    def all_central_frequencies(self):
+        return self.freq_dict
+            
+
+    @property
+    def gain(self):
+        return self.gain_dict
+
+
+    @gain.setter
+    def gain(self, val):
+        gain, channel= val
```

Comment:
Here you support a setter that takes a 2 element object with a gain value and channel. Earlier there was not setter for these objects and one needed to call `set_<parameter>(self, channel, value)`. I think I'm okay with either convention but I feel that it is important that it be uniform. It is also really important that the order of the unpacking be uniform throughout (`gain,ch = val` vs `ch,gain = val`) and should probably be in a doc string on these methods (I'd generally like to have more doc strings in all of the source, but it is really important when there are arguments which could be non-intuitive like those which need to be unpacked).

Less important, but I think that the reverse ordering is more intuitive (where the setting is going then the value to set). Others can disagree on that and the critical thing is only that it be consistent within the class.